packages/orchestrator-ui-components/src/components/WFOStartTaskButtonComboBox/WFOStartTaskButtonComboBox.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { useRouter } from 'next/router';
+
+import { useTranslations } from 'next-intl';
+import { useQueryWithGraphql } from '../../hooks';
+import { GET_WORKFLOWS_FOR_DROPDOWN_LIST_GRAPHQL_QUERY } from '../../graphqlQueries/workflows/workflowsQueryForDropdownList';
+import {
+    WorkflowComboBoxOption,
+    WFOButtonComboBox,
+} from '../WFOButtonComboBox/WFOButtonComboBox';
+import { PATH_START_WORKFLOW } from '../WFOPageTemplate';
+
+export const WFOStartTaskButtonComboBox = () => {
+    const router = useRouter();
+    const t = useTranslations('common');
+
+    const { data } = useQueryWithGraphql(
+        GET_WORKFLOWS_FOR_DROPDOWN_LIST_GRAPHQL_QUERY,
+        {
+            // Avoiding pagination by fetching an unrealistic amount of items
+            first: 1000,
+            after: 0,
+            filterBy: [{ field: 'target', value: 'SYSTEM' }],
+        },
+        'taskWorkflows',
+    );
+
+    const productList: WorkflowComboBoxOption[] = data?.workflows.page.map(
+        (w) => {
+            return {
+                label: w.description,
+                data: { workflowName: w.name, productId: '' },
+            };
+        },
+    ) as WorkflowComboBoxOption[];
```

Comment:
Are you sure you need this coersion? I would expect it to work without it

Reply:
+1

Reply:
Yes 100% sure

Reply:
Placed my comment at the wrong place, here in the correct context:
since `data` is possibly undefined, productList is also possibly undefined. A cast to WorkflowComboBoxOption[] is not recommended. Try to handle undefined or apply a fallback value as empty array: `?? []`

Reply:
Fixed and sorted